ios/insporation-share/ShareViewController.swift

```diff
@@ -0,0 +1,294 @@
+                    let userDefaults = UserDefaults(suiteName: "group.\(this.hostAppBundleIdentifier)")
+                    userDefaults?.set(this.sharedText, forKey: this.sharedTextKey)
+                    this.redirectToHostApp()
+                }
+                
+            } else {
+                self?.dismissWithError()
+            }
+        }
+    }
+    
+    private func handleUrl (content: NSExtensionItem, attachment: NSItemProvider, index: Int) {
+        
+        setSubjectText(content)
+        
+        attachment.loadItem(forTypeIdentifier: urlContentType, options: nil) { [weak self] data, error in
+            
+            if error == nil, let item = data as? URL, let this = self {
+                
+                this.sharedText.append(item.absoluteString)
+                
+                // If this is the last item, save imagesData in userDefaults and redirect to host app
+                if index == (content.attachments?.count)! - 1 {
+                    let userDefaults = UserDefaults(suiteName: "group.\(this.hostAppBundleIdentifier)")
+                    userDefaults?.set(this.sharedText, forKey: this.sharedUrlKey)
+                    this.redirectToHostApp()
+                }
+                
+            } else {
+                self?.dismissWithError()
+            }
+        }
+    }
+    
+    private func handleImages (content: NSExtensionItem, attachment: NSItemProvider, index: Int) {
+        
+        setSubjectText(content)
+        
+        attachment.loadItem(forTypeIdentifier: imageContentType, options: nil) { [weak self] data, error in
+            
+            if error == nil, let url = data as? URL, let this = self {
+                
+                // Always copy
+                let fileName = this.getFileName(from: url, type: .image)
+                let newPath = FileManager.default
+                    .containerURL(forSecurityApplicationGroupIdentifier: "group.\(this.hostAppBundleIdentifier)")!
+                    .appendingPathComponent(fileName)
+                let copied = this.copyFile(at: url, to: newPath)
+                if(copied) {
+                    this.sharedMedia.append(SharedMediaFile(path: newPath.absoluteString, thumbnail: nil, duration: nil, type: .image))
+                }
+                
+                // If this is the last item, save imagesData in userDefaults and redirect to host app
+                if index == (content.attachments?.count)! - 1 {
+                    let userDefaults = UserDefaults(suiteName: "group.\(this.hostAppBundleIdentifier)")
+                    userDefaults?.set(this.toData(data: this.sharedMedia), forKey: this.sharedMediaKey)
+                    this.redirectToHostApp()
+                }
+                
+            } else {
+                self?.dismissWithError()
+            }
+        }
+    }
+    
+    private func handleVideos (content: NSExtensionItem, attachment: NSItemProvider, index: Int) {
```

Comment:
What does this do? Diaspora doesn't allow to upload videos :)

Reply:
I remove this.